### PR TITLE
Fix a typo for computing the relative distance

### LINKED ISSUE #1822: first_intersection() doesn't always return the intersection closest to the point of the ray query
The code at [https://gist.github.com/adrian-bodenmann/1727c845bfa38cb29851a757dab44d61](https://gist.github.com/adrian-bodenmann/1727c845bfa38cb29851a757dab44d61) returns
`First intersection: 0.4 0.5 -0`
instead of 
`First intersection: 0.4 0.5 0.1`

Tested on Windows 7, Visual Studio 2013, CGAL 4.9

### PR DESCRIPTION
**Pull request by sloriot**
fixes #1822 and #1821 

cc @bo0ts Nice one ;)

### CODE REVIEW

File: Arrangement_on_surface_2/include/CGAL/Arr_polycurve_basic_traits_2.h [hunk 1]
```diff
@@ -870,6 +870,8 @@ class Arr_polycurve_basic_traits_2 {
       CGAL_precondition(i1 != INVALID_INDEX);
       CGAL_precondition(i2 != INVALID_INDEX);
 
+std::cout << "Inside Compare_y_at_x_right_2 " << cv1[i1] << " vs " << cv2[i2] << " at " << p << std::endl;
```
Comment (lrineau) [inline Arrangement_on_surface_2/include/CGAL/Arr_polycurve_basic_traits_2.h]: You patch is full of debug outputs

File: Sweep_line_2/include/CGAL/Sweep_line_2/Basic_sweep_line_2_impl.h [hunk 2]
```diff
@@ -452,14 +452,24 @@ void Basic_sweep_line_2<Tr, Vis, Subcv, Evnt, Alloc>::_sort_left_curves()
...
   CGAL_SL_PRINT_START_EOL("sorting left curves");
   CGAL_assertion(m_currentEvent->has_left_curves());
-
+  
   // Get the first curve associated with the event and its position on the
   // status line. We proceed from this position up the status line until
   // we encounter a subcurve that is not associated with the current event.
+  
+  CGAL_assertion(m_currentEvent->left_curves_begin() != m_currentEvent->left_curves_end());
+
```
Comment (lrineau) [inline Sweep_line_2/include/CGAL/Sweep_line_2/Basic_sweep_line_2_impl.h]: This PR should not modify anything but AABB tree.................

**Comment by bo0ts** [comment]
Ouch, why didn't this give an unused variable warning?

**Comment by sloriot** [comment]
No idea, both should indeed have.

File: AABB_tree/include/CGAL/internal/AABB_tree/AABB_ray_intersection.h [hunk 3]
```diff
@@ -187,10 +187,11 @@ class AABB_ray_intersection {
       typename AABB_traits::Geom_traits::Vector_3 v = ray->to_vector();
 
       for(int i = 0; i < 3; ++i) {
-        if(v[0] != FT(0.)) {
-          return x[0] / v[0];
+        if(v[i] != FT(0.)) {
+          return x[i] / v[i];
         }
```
Comment (lrineau) [inline AABB_tree/include/CGAL/internal/AABB_tree/AABB_ray_intersection.h]: Well, `i` is actually used: as the loop counter, in the expressions `i < 3` and `++i`. That is probably why no compiler warns about `i` being unused.

**Pull request merged**